src/services/waf-rules-services/list-waf-rules-service.js

```diff
@@ -0,0 +1,82 @@
+import { AxiosHttpClientAdapter, parseHttpResponse } from '../axios/AxiosHttpClientAdapter'
+import { makeWafRulesBaseUrl } from './make-waf-rules-base-url'
+
+export const listWafRulesService = async () => {
+  let httpResponse = await AxiosHttpClientAdapter.request({
+    url: `${makeWafRulesBaseUrl()}/?page=1&page_size=100`,
+    method: 'GET'
+  })
+
+  httpResponse = adapt(httpResponse)
+
+  return parseHttpResponse(httpResponse)
+}
+const parseStatusData = (status) => {
+  const parsedStatus = status
+    ? {
+        content: 'Active',
+        severity: 'success'
+      }
+    : {
+        content: 'Inactive',
+        severity: 'danger'
+      }
+
+  return parsedStatus
+}
+const adapt = (httpResponse) => {
+  /**
+   * Necessary until the API gets the common pattern
+   * of returning the array of data inside results property
+   * like other andpoints.
+   */
+
+  const threatTypes = {
+    file_upload: 'File upload',
+    evading_tricks: 'Evading Tricks',
+    unwanted_access: 'Unwanted Access',
+    identified_attack: 'Identified Attack',
+    cross_site_scripting: 'Cross-Site Scripting (XSS)',
+    directory_traversal: 'Directory Traversal',
+    remote_file_inclusion: 'Remote File Inclusions (RFI)',
+    sql_injection: 'SQL Injection'
+  }
+
+  const isArray = Array.isArray(httpResponse.body.results)
+
+  const parsedWafRules = isArray
+    ? httpResponse.body.results.map((waf) => {
+        const threatTypesArray = []
+        for (const key in threatTypes) {
+          if (Object.hasOwnProperty.call(threatTypes, key)) {
```

Comment:
sugestão de return nos casos de false para evitar if aninhado em for.
const hasThreatType = Object.hasOwnProperty.call(threatTypes, key) // parece que sem vai ter.
 como threatTypes gera cada key, Object.hasOwnProperty vai sempre dar true não?